webpack.config.js

```diff
@@ -1,21 +1,3 @@
-const path = require('path');
-
-module.exports = {
-  entry: path.resolve(__dirname, './src/index.js'),
-  module: {
-    rules: [
-      {
-        test: /\.(js|jsx)$/,
-        exclude: /node_modules/,
-        use: ['babel-loader'],
-      },
-    ],
-  },
-  resolve: {
-    extensions: ['*', '.js', '.jsx'],
-  },
-  output: {
-    path: path.resolve(__dirname, './public'),
-    filename: 'bundle.js',
-  },
-};
+module.exports = process.env.NODE_ENV === 'development'
+    ? require('./webpack.dev.config.js')
+    : require('./webpack.prod.config.js');
```

Comment:
try to make use of 'webpack-merge' module to meet evaluation criteria for this task.
You need to have 'common', 'dev', 'prod' configs. 

You have ` webpack.config.js` as entry point and there you check whether it dev or prod and use something like like `merge(common, dev)`.
In this case common config has all the settings which are common for both dev and prod, and dev/prod configs have only env related settings